File: HighestAltitude.java
package codenights;

import java.util.*;

public class HighestAltitude {

	public static void main(String[] args) {
		int gain[] = {-4,-3,-2,-1,4,3,2};
		System.out.println(largestAltitude(gain));
	}
	public static int largestAltitude(int[] gain) {

		int max = gain[0], pos = 0;

		for( int x : gain) {

			if( (pos + x) > max)
				max = pos + x;

			pos += x;   
		}

		if(max < 0)
			return 0;
		else
			return max;
	}

}

File: highestAltitude.js
/**
 * Leetcode: 1732. Find the Highest Altitude
 * @param {number[]} gain
 * @return {number}
 */
var largestAltitude = function(gain) {
    //max
    //altitude = 0
    //loop gain
    //  max = max(max, altitude + gain[i] )
    //  altitude += gain[i]
    //return max

    let max = altitude = 0;

    for(let i =0; i< gain.length; i++){
        max = Math.max(max, altitude + gain[i]);
        altitude += gain[i];
    }

    return max < 0 ? 0 : max;

};